Automattic/wp-calypso
# Add loading state while plans are being loaded from backend

Prevents wrong window to be displayed when loading plans



Related to https://github.com/orgs/Automattic/projects/188#card-88397144
Fixes #74773 

## Proposed Changes

* Add a loading state
* Design discussions: p1679489823677779-slack-C02DQP0FP

## Testing Instructions



* Build this with `yarn start`
* Go to http://calypso.localhost:3000/earn/payments-plans/testproductionnewsletter.wordpress.com
* And you should see a proper state transition
* Feel free to check if the rest is loading the right way as well

<img width="993" alt="Screenshot 2023-03-22 at 20 55 43" src="https://user-images.githubusercontent.com/790558/226911353-afb852f1-7c06-420c-a3ef-1c48955a59d5.png">

## Pre-merge Checklist
